controllers/service/service_controller.go

```diff
@@ -0,0 +1,99 @@
+package service
+
+import (
+	"context"
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/aws-alb-ingress-controller/controllers/service/eventhandlers"
+	"sigs.k8s.io/aws-alb-ingress-controller/pkg/annotations"
+	"sigs.k8s.io/aws-alb-ingress-controller/pkg/k8s"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	LoadBalancerFinalizer   = "service.k8s.aws/load-balancer-finalizer"
+	ServiceAnnotationPrefix = "service.beta.kubernetes.io"
+	controllerName          = "service"
+)
+
+func NewServiceReconciler(k8sClient client.Client, log logr.Logger) *ServiceReconciler {
+	return &ServiceReconciler{
+		k8sClient:        k8sClient,
+		log:              log,
+		annotationParser: annotations.NewSuffixAnnotationParser(ServiceAnnotationPrefix),
+		finalizerManager: k8s.NewDefaultFinalizerManager(k8sClient, log),
+	}
+}
+
+type ServiceReconciler struct {
+	k8sClient        client.Client
+	log              logr.Logger
+	annotationParser annotations.Parser
+	finalizerManager k8s.FinalizerManager
+}
+
+// +kubebuilder:rbac:groups="",resources=services,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups="",resources=services/status,verbs=get;update;patch
+func (r *ServiceReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+	_ = r.log.WithValues("service", req.NamespacedName)
```

Comment:
i think we can remove this line if the return value `_ = r.log.WithValues("service", req.NamespacedName)` is not used.